src/main/java/com/openlattice/datastore/data/controllers/DataController.java

```diff
@@ -712,37 +713,61 @@ public Integer deleteEntityProperties(
     }
 
     private List<WriteEvent> clearAssociations( UUID entitySetId, Optional<Set<UUID>> entityKeyIds ) {
+        final Map<UUID, Map<UUID, PropertyType>> authorizedPropertyTypes = new HashMap<>();
+
         // collect association entity key ids
         final PostgresIterable<DataEdgeKey> associationsEdgeKeys = collectAssociations( entitySetId, entityKeyIds );
 
-        // access checks
-        final Map<UUID, Map<UUID, PropertyType>> authorizedPropertyTypes = new HashMap<>();
-        associationsEdgeKeys.forEach( edgeKey -> {
+        // collect edge entity sets
+        final var edgeEntitySetIds = associationsEdgeKeys.stream()
+                .map( edgeKey -> edgeKey.getEdge().getEntitySetId() )
+                .collect( Collectors.toSet() );
+        final var edgeEntitySets = edmService.getEntitySetsAsMap( edgeEntitySetIds );
```

Comment:
It would also be lighter to use a hazelcast aggregator to filter to a set of entity set ids that match the audit edge entity type instead of loading the full entity set objects

Reply:
Yeah, right, will do!